openupgrade_scripts/scripts/website_sale/16.0.1.1/post-migration.py

```diff
@@ -0,0 +1,17 @@
+from openupgradelib import openupgrade
+
+
+def login_or_registration_required_at_checkout(env):
+    """ In case the website_sale_require_login module was installed in v15,
+    the value of the account_on_checkout field is set to "mandatory" to make
+    login/registration mandatory for checkout. """
+    if env["ir.module.module"].search(
+        [("name", "=", "website_sale_require_login")]
+    ).state == "installed":
```

Comment:
Is it working ? 
Afaik, the renaming of the module (website_sale_require_login -> website_sale) is done in the pre-migration script of the base module, here : https://github.com/OCA/OpenUpgrade/blob/16.0/openupgrade_scripts/scripts/base/16.0.1.3/pre-migration.py#L106

So, in the post migration step of the website_sale module, the website_sale_require_login is not installed, as the entry has been dropped by https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L1776

Did I missed something ?

Reply:
I was making changes for the pre-migration to run and just pushed them :) that wasn't going to work, thanks!

Reply:
>> So, in the post migration step of the website_sale module, the website_sale_require_login is not installed, as the entry has been dropped by https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L1776

I had doubts if the record stayed in the database in post, @pedrobaeza has been telling me about it and this clears it up for me, thanks again!

Reply:
the pre-migration of website_sale, is executed after the pre-migration of the base module. So, same problem, I think that it will return always false.

Note : an helper function exist https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L2521

Reply:
The maybe we could take it out from the apriori and flag it to uninstall in the migration script. What do you think?

Reply:
So I understand that in the case of a merge module there is no way to know if it was installed or not, I will continue investigating, thank you very much.

Reply:
@chienandalu That's what I was thinking. I think it's the only option to be able to have the data and do the check. Thanks!

Reply:
Do these operations in `base` pre-migration before the call of `update_module_names`

Reply:
Finally, the merge is not indicated in apriori, the appropriate operations are performed in pre-migration and the "update_module_names" method is called to perform the module merge. Correct me if it is not correct please.

Reply:
@pedrobaeza  I think we were writing at the same time, wouldn't it be correct to call the "update_module_names" method from the script itself? is it better to do it as you say?

Reply:
That can be another option, yes. As you prefer.

Reply:
Hi @pilarvargas-tecnativa, thanks for the changes.

The implementation you propose is perfectly valid.

However, I'm a bit disappointed if the apriori file doesn't contain all the module merges.
I find it interesting to have a single file with all the changes from one version to another. It's very readable by any people, and it can also be parsed easily by a script.

Do you think we could implement Pedro's solution instead (do it in the "base" module)?

Reply:
@legalsylvain you are right. I have made the changes and will keep this in mind for the next time I am faced with something similar. Thank you.